Subject: [PATCH] gitweb: include timestamp day in ()

It's very misleading to just show the actual HH:MM %z in parentheses
without the day because the "localized" time being shown may have
an entirely different day.

Always include the day of the month in parentheses as well.

Signed-off-by: Kyle J. McKay <mackyle@gmail.com>
---
 gitweb/gitweb.perl | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/gitweb/gitweb.perl b/gitweb/gitweb.perl
index c34fc944..607e881b 100755
--- a/gitweb/gitweb.perl
+++ b/gitweb/gitweb.perl
@@ -3759,6 +3759,7 @@ sub parse_date {
 	($sec, $min, $hour, $mday, $mon, $year, $wday, $yday) = gmtime($local);
 	$date{'hour_local'} = $hour;
 	$date{'minute_local'} = $min;
+	$date{'mday_local'} = $mday;
 	$date{'tz_local'} = $tz;
 	$date{'iso-tz'} = sprintf("%04d-%02d-%02d %02d:%02d:%02d %s",
 	                          1900+$year, $mon+1, $mday,
@@ -5040,12 +5041,12 @@ sub format_timestamp_html {
 		$strtime = qq!<span class="$datetime_class">$strtime</span>!;
 	}
 
-	my $localtime_format = '(%02d:%02d %s)';
+	my $localtime_format = '(%d %02d:%02d %s)';
 	if ($date->{'hour_local'} < 6) {
-		$localtime_format = '(<span class="atnight">%02d:%02d</span> %s)';
+		$localtime_format = '(%d <span class="atnight">%02d:%02d</span> %s)';
 	}
 	$strtime .= ' ' .
-	            sprintf($localtime_format,
+	            sprintf($localtime_format, $date->{'mday_local'},
 	                    $date->{'hour_local'}, $date->{'minute_local'}, $date->{'tz_local'});
 
 	return $strtime;
---
